app/src/main/java/pl/hypeapp/wykopolka/presenter/DemandQueuePresenter.java

```diff
@@ -72,6 +72,56 @@ public void onTransferBook(int position) {
         getView().showTransferDialog(bookPendingUserPair);
     }
 
+    public void onNotificationAction(Pair<Book, PendingUser> bookPendingUserPair) {
+        if (bookPendingUserPair.second.getWasCalled()) {
+            showMikroblogEntry(bookPendingUserPair.second.getCallId());
+        } else {
+            notifyUserOnMikroblog(bookPendingUserPair.second.getWishId());
+        }
+    }
+
+    private void showMikroblogEntry(String callId) {
+        getView().showMikroblogEntry(callId);
+    }
+
+    private void notifyUserOnMikroblog(String wishId) {
+        getView().setDialogInLoading();
+        String sign = HashUtil.generateApiSign(mAccountKey, wishId);
+        rxHelper.manageSubscription(mWykopolkaApi.callUserOnMikroblog(mAccountKey, wishId, sign)
+                .compose(RxTiPresenterUtils.<ResponseBody>deliverToView(this))
+                .subscribeOn(Schedulers.newThread())
```

Comment:
Repeated part of code. You should put `subscribeOn` and `observerOn` method to specify class like `RxTransofrmerProvider` and you should apply schedulers by `compose` method.